oeps/oep-0002-bp-repo-metadata.rst

```diff
@@ -37,19 +37,28 @@ Bear in mind that particular classes of Open edX repositories may have different
 Specification
 =============
 
+Keys
+----
+
 Each repo will include a file ``openedx.yaml``, with the following keys:
 
-``owner``: string (required unless ``archived`` is ``True``)
-    The GitHub user responsible for the repository.
+``owner``: dictionary (required)
+    This key contains information about the assigned owner of this repository.
+
+    ``type``: string (required)
+        The value of this key must be either ``team`` or ``repo``. It indicates which ownership model applies to this repository, and therefore which of these two keys should exist with a non-empty value.
 
-``supporting_teams``: list of strings (optional)
-    The GitHub teams supporting the owner for this repository.
+    ``team``: string (optional)
+        The GitHub team responsible for this repository, prefixed with the Github organization (e.g., ``edx/arch-bom``). If this value is set, this repository must have one of the following Tags_: ``webservice``, ``frontend-app``, or ``framework``. That is, this repository contains high-level, business-aligned functionality that is suitable for direct ownership designation to an individual responsible team.
+
```

Comment:
It's not obvious to me that a "team" owner can only apply to webservice, frontend-app, or framework.  I don't think "must" for the tags is helpful here.  Wouldn't repo-tools be owned by the Open edX team? It's not one of those three things.

Reply:
The proposal is that `repo-tools` would have an `owner.type` of `repo` and that `repo` would be a higher-order webservice, frontend-app, or framework.  In this case, it can be the `openedxstats` webservice.

Do you have an alternative proposal that works better, while serving our goal of finding a more maintainable ownership assignment model?

Reply:
repo-tools gets used by a few other repos.  repo-tools-data even more so.  A library repo is likely to be used by more than one other repo, isn't that one of the reasons it's been made into a library?

If a team is willing to say, "We own this library," why should this OEP forbid it?  Perhaps most libraries do fit into the "owned by a repo" model, great.

Reply:
As another example: `edx_lint` is a library (and commandline tool) used by a number of other repos. However, it is none of `webservice`, `frontend-app`, or `framework`, and also isn't appropriate to be owned by another repo.

Reply:
Just one more example: edx-drf-extensions is used by a few different IDAs.

Reply:
The intention here is to _not_ assign teams/owners to low-level technical repositories.  That results in harder to maintain assignments as the platform and teams evolve and also makes it harder to keep track of these in our conversations with Product.  Rather, we'd like to have the mapping between "humans" and "technology" at higher-level concepts.

To support this, we will need to make technical decisions on (1) the single-responsibilities of repos and (2) map lower-level repos with higher-order technical concerns.  Here are some examples:
* `edx-drf-extensions` can be listed as a `framework` related to APIs and renamed to something like `edx-rest-api-providers`. (It is listed below as an example `framework`.)
* `auth-backends` can be listed as a `library` related to user-authentication and owned by the `user_authn` app owner within edx-platform.
* `edx_lint` can be its own `framework` related to linting or a `library` mapped to a higher-order framework - depending on what we find in our exercise to assign all the repos.

The _proof is in the pudding_.  As we go through the effort of comprehensively assigning repos, we'll see how the current proposal for tags plays out.  If needed, we can add/edit.  The primary motive here is to assign ownership to humans at higher-order concepts.

Reply:
I understand the goal of trying to assign ownership in ways that ease maintenance.  I don't think calling edx-lint a framework is a good way to accomplish that. It's not a framework. It's a tool and a library. Why can't a library be owned by a team? Most libraries won't, but some will.

Why not simply say, "as much as possible, repos should be owned by other repos, and that the uppermost repos should be owned by teams"? That's easily understood. You can provide guidance that many libraries naturally fit into ownership by another repo.  It feels arbitrary to say that "frameworks" are top-level and "libraries" are not. It feels contorted to call edx-lint a framework just so that we can allow it to be owned by a team.

Reply:
That's fair.  I'll update the OEP to not be restrictive of which `tags` imply team-ownership. This decouples ownership from repo types.

Do note though, `edx-lint` doesn't abide by the current definition of a `library`:  `Code runs in production..`

Once again, I do expect us to update this OEP again once we comprehensively update all the repos and see what additional tags are missing and needed.

Reply:
A `tools` tag might be warranted.